substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/SubstrateLLVMBackend.java

```diff
@@ -76,14 +76,14 @@ public SubstrateLLVMBackend(Providers providers) {
     }
 
     @Override
-    public CompilationResult createJNITrampolineMethod(ResolvedJavaMethod method, CompilationIdentifier identifier,
-                    RegisterValue threadArg, int threadIsolateOffset, RegisterValue methodIdArg, int methodObjEntryPointOffset) {
+    public CompilationResult createJNITrampolineMethod(ResolvedJavaMethod method, CompilationIdentifier identifier, int threadIsolateOffset,
+                                                       boolean nonVirtual, int methodObjEntryPointOffset, CallingConvention callingConvention) {
 
         CompilationResult result = new CompilationResult(identifier);
         result.setMethods(method, Collections.emptySet());
 
         LLVMGenerator generator = new LLVMGenerator(getProviders(), result, null, method, 0);
-        generator.createJNITrampoline(threadArg, threadIsolateOffset, methodIdArg, methodObjEntryPointOffset);
+        generator.createJNITrampoline(threadIsolateOffset, nonVirtual ? 3 : 2, methodObjEntryPointOffset);
```

Comment:
Do not use raw integers that way. The parameter indices should be computed when creating the parameter list and passed to the method.